community/detectors/apache_druid_preauth_rce_cve_2021_25646/src/main/java/com/google/tsunami/plugins/detectors/rce/cve202125646/ApacheDruidPreAuthRCECVE202125646VulnDetector.java

```diff
@@ -0,0 +1,164 @@
+
+/**
+ * A {@link VulnDetector} that detects the CVE-2021-25646 vulnerability.
+ */
+@PluginInfo(
+    type = PluginType.VULN_DETECTION,
+    name = "ApacheDruidPreAuthRCECVE202125646VulnDetector",
+    version = "1.0",
+    description = "This detector checks for Apache Druid <= 0.20.0 CVE-2021-25646 "
+        + "Pre-Auth RCE vulnerability.",
+    author = "threedr3am (qiaoer1320@gmail.com)",
+    bootstrapModule = ApacheDruidPreAuthRCECVE202125646VulnDetectorBootstrapModule.class
+)
+public class ApacheDruidPreAuthRCECVE202125646VulnDetector implements VulnDetector {
+
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+
+  private final Clock utcClock;
+  private final HttpClient httpClient;
+
+  private final String payloadString;
+  private static final String CHECK_VUL_PATH = "druid/indexer/v1/sampler";
+
+  @Inject
+  ApacheDruidPreAuthRCECVE202125646VulnDetector(@UtcClock Clock utcClock, HttpClient httpClient) {
+    this.utcClock = checkNotNull(utcClock);
+    this.httpClient = checkNotNull(httpClient);
+    String payloadString;
+    try {
+      payloadString = Resources.toString(
+          Resources.getResource(this.getClass(), "payloadString.json"), UTF_8);
+    } catch (IOException e) {
+      throw new AssertionError("Couldn't load payload resource file.", e);
+    }
+    this.payloadString = payloadString;
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(this::isWebServiceOrUnknownService)
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private boolean isWebServiceOrUnknownService(NetworkService networkService) {
+    return networkService.getServiceName().isEmpty()
+        || NetworkServiceUtils.isWebService(networkService)
+        || NetworkServiceUtils.getServiceName(networkService).equals("unknown");
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    HttpHeaders httpHeaders = HttpHeaders.builder()
+        .addHeader(com.google.common.net.HttpHeaders.CONTENT_TYPE, MediaType.JSON_UTF_8.toString())
+        .build();
+
+    ByteString requestBody = ByteString.copyFromUtf8(payloadString);
+    String targetUri = buildTargetUrl(networkService);
+    try {
+      HttpResponse response = httpClient.send(
+          post(targetUri).setHeaders(httpHeaders).setRequestBody(requestBody).build(),
+          networkService);
+      if (response.status() == HttpStatus.OK && response.bodyString().isPresent()) {
+        String responseBody = response.bodyString().get();
+        if (responseBody.contains("JavaScript is disabled")) {
+          return false;
+        }
```

Comment:
It looks like you don't actually need this check. The `if` block immediately after this code implies the return value for this condition to be `false`.